### PR TITLE
Setup Github workflow

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Signed-off-by: Lixia (Sylvia) Lei <lixia_lei@outlook.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Go ${{ matrix.go-version }} environment
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+          check-latest: true
+      - name: Run unit tests
+        run: make test
```
Comment (gokarnm) [inline .github/workflows/build.yml:25]: I'm curious why the steps across [notation](https://github.com/notaryproject/notation/blob/main/.github/workflows/build.yml#L23-L34), [notation-go](https://github.com/notaryproject/notation-go/pull/60/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R25-R33), and notation-core-go are slightly different, can they all be the same set of steps. Can they be [reused workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#reusable-workflows-and-starter-workflows) defined in a single place?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (gokarnm) [inline .github/workflows/build.yml:11]: We should update notation [build.yml](https://github.com/notaryproject/notation/blob/main/.github/workflows/build.yml#L11) to use the same.

Comment (rgnote) [reply .github/workflows/build.yml:11]: Can we also build/test on Windows and Mac?

Comment (Wwwsylvia) [reply .github/workflows/build.yml:11]: Make sense. But we may want to do this in a seperate PR, as currently we use make test to generate coverage report, which will not work on Windows.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Wwwsylvia) [reply .github/workflows/build.yml:25]: Good idea!

File: .github/workflows/build.yml (hunk 2 above)
Comment (Wwwsylvia) [reply .github/workflows/build.yml]: @gokarnm Updated notation workflow. See https://github.com/notaryproject/notation/pull/212/files

File: .github/workflows/build.yml (hunk 1 above)
Comment (Wwwsylvia) [reply .github/workflows/build.yml]: Defined a reusable workflow.

**Review by gokarnm — APPROVED** [review]
LGTM!

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**